src/main/httpCommandServer.ts

```diff
@@ -0,0 +1,108 @@
+import fs from 'fs';
+import http from 'http';
+import path from 'path';
+
+import Logging from '@/utils/logging';
+import paths from '@/utils/paths';
+
+type ServerState = {
+  user: string;
+  password: string;
+  port: number;
+  pid: number;
+}
+
+const console = Logging.server;
+const ServerPort = 6107;
+const ServerUsername = 'user';
+const ServerFileBasename = 'rd-engine.json';
+
+export default class HttpCommandServer {
+  protected server: http.Server = http.createServer();
+  protected username = ServerUsername;
+  protected password = randomStr();
+  protected stateInfo: ServerState = {
+    user:     this.username,
+    password: this.password,
+    port:     ServerPort,
+    pid:      process.pid,
+  };
+
+  async init() {
+    const statePath = path.join(paths.appHome, ServerFileBasename);
+
+    await fs.promises.writeFile(statePath,
+      JSON.stringify(this.stateInfo, undefined, 2),
+      { mode: 0o600 });
+    this.server.listen(ServerPort);
+    console.log(`Listening on port ${ ServerPort }, user: ${ ServerUsername },  password: ${ this.password }`);
```

Comment:
Don't write `password` to the log. Console log may get redirected to a file, and including the password negates the point of saving `rd-engine.json` with file mode `0o600`.

Reply:
Pulled it out. Wondering if we should write out the username for a similar reason. If this server ever supports multiple users we don't want their names to be discoverable either.

Reply:
I don't see any value in logging the username. I don't think we will ever support multiple users; this is just a secret token to protect the API. We only use basic auth because that is easy to use with common tools like `curl`.